pystiche/papers/ulyanov_et_al_2016/data.py

```diff
@@ -0,0 +1,273 @@
+        ),
+        "the_scream": DownloadableImage(
+            urljoin(style_base_johnson, "the_scream.jpg"),
+            md5="619b4f42c84d2b62d3518fb20fa619c2",
+        ),
+        "Jean Metzinger": DownloadableImage(
+            "https://upload.wikimedia.org/wikipedia/commons/c/c9/Robert_Delaunay%2C_1906%2C_Portrait_de_Metzinger%2C_oil_on_canvas%2C_55_x_43_cm%2C_DSC08255.jpg",
+            title="Portrait of Jean Metzinger",
+            author="Jean Metzinger",
+            date="1906",
+            license=PublicDomainLicense(1906),
+            md5="3539d50d2808b8eec5b05f892d8cf1e1",
+            file="jean_metzinger.jpg",
+        ),
+        "mosaic": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "mosaic.jpg"),
+            md5="4f05f1e12961cebf41bd372d909342b3",
+        ),
+        "pleades": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "pleades.jpg"),
+            md5="6fc41ac30c2852a5454a0ead2f479dc9",
+        ),
+        "starry": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "starry.jpg"),
+            md5="c6d94f7962466b2e80a64ae82523242a",
+        ),
+        "turner": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "turner.jpg"),
+            md5="7fdd9603a5182dcef23d7fb1c5217888",
+        ),
+    }
+    return DownloadableImageCollection(
+        {**texture_images, **content_images, **style_images},
+        root=root,
+        download=download,
+        overwrite=overwrite,
+    )
+
+
+def ulyanov_et_al_2016_dataset(
+    root: str,
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    transform: Optional[Transform] = None,
+):
+    if transform is None:
+        transform = ulyanov_et_al_2016_content_transform(
+            impl_params=impl_params, instance_norm=instance_norm
+        )
+    return ImageFolderDataset(root, transform=transform)
+
+
+def ulyanov_et_al_2016_batch_sampler(
+    data_source: Sized,
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    stylization: bool = True,
+    num_batches=None,
+    batch_size=None,
+) -> FiniteCycleBatchSampler:
+
+    if num_batches is None:
+        if impl_params:
+            if instance_norm:
+                num_batches = 2000
+            else:
+                num_batches = 300 if stylization else 150
+        else:
+            num_batches = 200
+
+    if batch_size is None:
+        if impl_params:
+            if instance_norm:
+                batch_size = 1
+            else:
+                batch_size = 4 if stylization else 16
+        else:
+            batch_size = 16
+
+    return FiniteCycleBatchSampler(
+        data_source, num_batches=num_batches, batch_size=batch_size
+    )
+
+
+def ulyanov_et_al_2016_image_loader(
+    dataset: Dataset,
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    stylization: bool = True,
+    batch_sampler: Optional[Sampler] = None,
+    num_workers: int = 4,
```

Comment:
What is the default value for `num_workers` for `DataLoader`? We should also use this here. Using `4` seems arbitrary.

Reply:
He uses threads for his `dataloader` in the branch `master`, where the default setting is `nthreads = 4`. Is this not the same as in this case `num_workers`?

Reply:
Yes, that should be the same but since this has only an effect on the training speed and not the performance we should stick to the default value which is `0`.